### PR TITLE
UI: Update searching of oscap entities according to #2930

### PR DESCRIPTION
**Pull request by abalakh**
Part of #2930

### CODE REVIEW

**Comment by abalakh** [comment]
I wrote simple test to verify oscap policies:
```python
def test_policies(self):
        with Session(self.browser) as session:
            name = gen_string('alphanumeric')
            make_oscappolicy(
                session,
                name=name,
                period='Monthly',
                period_value='10'
            )
            self.assertIsNotNone(self.oscappolicy.search(name))
            self.oscappolicy.delete(name)
            self.assertIsNone(self.oscappolicy.search(name))
```
Test results:
```
nosetests tests/foreman/ui/test_openscap.py -m test_policies
.
----------------------------------------------------------------------
Ran 1 test in 77.828s

OK
```

File: robottelo/ui/oscapcontent.py [hunk 1]
```diff
@@ -8,6 +8,15 @@
 
 class OpenScapContent(Base):
     """Manipulates OpenScap content from UI"""
+    search_key = 'login'
```
Comment (oshtaier) [inline robottelo/ui/oscapcontent.py]: why 'login'?

File: robottelo/ui/oscappolicy.py [hunk 2]
```diff
@@ -9,8 +9,16 @@
 class OpenScapPolicy(Base):
     """Manipulates OpenScap Policy from UI"""
 
+    def navigate_to_entity(self):
+        """Navigate to OpenScap Policy entity page"""
+        Navigator(self.browser).go_to_oscap_policy()
+
+    def _search_locator(self):
+        """Specify locator for OpenScap Policy entity search procedure"""
+        return locators['oscap.select_policy']
```
Comment (oshtaier) [inline robottelo/ui/oscappolicy.py]: `common_locators['search']` was used before. Is it specific reason for that new locator?

**Comment by oshtaier** [comment]
Pending questions and at least few tests for all these functionalities

File: robottelo/ui/oscapcontent.py (hunk 1 above)
Comment (abalakh) [inline robottelo/ui/oscapcontent.py]: :+1:

File: robottelo/ui/oscappolicy.py (hunk 2 above)
Comment (abalakh) [inline robottelo/ui/oscappolicy.py]: And `common_locators['search']` will be used in `Base.search()`. This is `element_locator`, not searchbar locator. And btw test result prooves it's working (at least for policies).

**Comment by abalakh** [comment]
Comments addressed. Regarding test results - there's 1 test for policies which verifies create/search/delete pretty fine.
I've talked to @kbidarkar regarding content and reports, there's no easy way to verify them as we have only 1 longrun test, which will run for at least 1 hour and require a lot of configurational preconditions. He also said that this functionality is still under development and searching and deletion may not even work as for now (i even fixed some of bugs in this PR). So i propose to merge PR as it follows our new search approach and introduces some fixes (at least for policies search/delete and content delete for sure).

File: robottelo/ui/oscappolicy.py (hunk 2 above)
Comment (oshtaier) [inline robottelo/ui/oscappolicy.py]: Nice, so previous search method was looking for unknown things :)

**Comment by oshtaier** [comment]
ACK. Seems logical as for me

**Comment by kbidarkar** [comment]
Small correction: May be I was not fully clear, it was under development and the code was right, but was not fully completed. So `searching` would work but there is no easy way of testing it right away.

Also `Delete` at this point was not even tested, meaning no test-cases were added.

**Comment by kbidarkar** [comment]
Here is the Trello card for it, "In Progress". https://trello.com/c/LVyz6W1K/17-oscap

File: robottelo/ui/locators.py [hunk 3]
```diff
@@ -2101,6 +2101,10 @@ def __iter__(self):
         By.XPATH, "//input[contains(@data-url, 'auto_complete_search')]"),
     "oscap.content_edit": (
         By.XPATH, "//td[contains(.,'%s')]/../td/div/span/a"),
+    "oscap.content_dropdown": (
+        By.XPATH,
+        ("//td[contains(.,'%s')]/following-sibling::td/div/"
+         "a[@data-toggle='dropdown']")),
     "oscap.content_download": (
```
Comment (kbidarkar) [inline robottelo/ui/locators.py]: :+1:

File: robottelo/ui/locators.py [hunk 4]
```diff
@@ -2141,6 +2145,8 @@ def __iter__(self):
     "oscap.profile_policy": (By.ID, "policy_scap_content_profile_id"),
     "oscap.period_policy": (By.ID, "policy_period"),
     "oscap.weekday_policy": (By.ID, "policy_weekday"),
+    "oscap.dayofmonth_policy": (By.ID, "policy_day_of_month"),
+    "oscap.custom_policy": (By.ID, "policy_cron_line"),
```
Comment (kbidarkar) [inline robottelo/ui/locators.py]: :+1:

**Comment by elyezer** [comment]
ACK

**Comment by kbidarkar** [comment]
Thanks to @abalakh for updating many stuff related to OpenScap. :+1: 

ACK

**Pull request merged**